src/ScottPlot4/ScottPlot/Drawing/Colormap.cs

```diff
@@ -151,6 +151,32 @@ public static int[] GetRGBAs(double[] intensities, Colormap colorMap, double min
             return rgbas;
         }
 
+        public static int[] GetRGBAs(double[] intensities, double[] alphaVals, Colormap colorMap)
+        {
+            int[] rgbas = new int[intensities.Length];
+            for (int i = 0; i < intensities.Length; i++)
+            {
+                byte pixelIntensity = (byte)Math.Max(Math.Min(intensities[i] * 255, 255), 0);
+                var (r, g, b) = colorMap.GetRGB(pixelIntensity);
+                byte alpha = (byte)Math.Max(Math.Min(alphaVals[i] * 255, 255), 0);
```

Comment:
I think this is equivalent to the simpler `Math.Clamp(alphaVals[i] * 255, 0, 255)`, however, it may be simpler to just take in `alphaVals` as a `byte[]` instead, seeing as we only use it as a byte. This would also use 1/8 as much memory.

I think this applies everywhere but I just wrote it the once.

Reply:
Note that `Math.Clamp()` is only available in .NET 6 and newer, so we can't use it since we target .NET Framework 😝

Reply:
> Looks good, potentially an option to simplify things by using `byte[]` as an alpha mask rather than `double[]`

This is an interesting choice... I guess it comes down to whether the heatmap API is more like an _image_ or if that concept is abstracted away and we interact with the heatmap more like _data_.

Consider these alternative names for the `alphaVals` argument in `GetRGBAs()`: 

* When regarding heatmap values as image pixel intensities, `byte[,]? alpha` makes sense with a range 0-255

* When regarding heatmap values as data measurements, `double[,]? opacity` makes sense with a range 0-1

I think I'm leaning toward the second option because it's a little more easy to reason about when plotting non-image data as heatmaps. Sure it's converted to images _under the hood_, but I think the heatmap API should abstract that complexity away from the users who don't need to think about it.

Reply:
I would agree, but the parameter is called `alphaVals`, so we're clearly alluding to an image's alpha channel.

It's also worth noting that if a user is working with colours, they're working with `System.Drawing.Color`, which exposes itself as a struct of four 8-bit colour channels, so dealing with a fractional opacity might even be less intuitive to users.

Reply:
I think we're saying the same thing.

* If we keep calling it "alpha", `byte` makes sense with a range 0-255

* If we want to change it to "opacity", `double` makes sense with a range 0-1

My lean is toward the second one, which seems to be to me a little more consistent with `double[] intensities` in being more data-ish and less image-ish

Also I recognize this is splitting hairs 🤷 and I'm happy with either outcome. I'll wait to hear @bukkideme's thoughts and after this conversation is resolved I'm looking forward to merging this in 🚀

Reply:
Hi! Actually i can agree and reason with both sides. I think it is a bit semantics question too. 

But what about if we rename the names "alpha" to "opacity" everywhere, and keep it double? Even if it is reduced to bytes in this very implementation. We work with data which has more meaning than only the visualizable alpha channel, so keeping it double serves us better. For example, such mask matrix can be used really as double when we use it for data manipulation. 
Another thing, software Gwyddion also defines masks as double 2D arrays, even if using it as a binary 0/1 matrix in present version. Keeping it double makes it more future proof.

Reply:
> I can agree and reason with both sides

We have two good ideas here! I'll step in and pick one and merge this PR, and we can always change it later if a strong reason to arises.

I'm going to resolve this conversation and merge now, but we can continue the discussion or additional modifications in new PRs as needed

> Keeping it double makes it more future proof

I like this plan!